### DISCUSSION TITLE
On desktop, the enter key in the shopping list should add the item to the list and open a new item, with the textbox focused; instead of adding a newline in the textbox.

### DISCUSSION THREAD

**Discussion by Fastjur** [mealie-recipes/mealie#3114] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

The shopping list feature is a bit clunky to use on desktop, as it requires one to use the mouse to click multiple times on various (small) buttons when one wants to add new items to the shopping list.
I can count on one hand the amount of times I need linebreaks in my shopping list entries, but the number of times I just quickly want to add 10+ items to a shopping list is quite high.

GIF of unwanted behaviour, notice how many mouse movements and clicks are required for every item:
![chrome_gxKBcSJHMv](https://github.com/mealie-recipes/mealie/assets/1493561/6d70b997-a571-4666-b591-973ba54a16b2)

### Please provide a concise description of the feature that would resolve your issue.

The enter key should, when the textbox is focused when adding a new item, should by default save the item to the list and immediately open a new item, such that it is much faster to add multiple items.

An example from google keep with this behaviour:
![chrome_aRkxfNgtVs](https://github.com/mealie-recipes/mealie/assets/1493561/5bb66d17-0d11-4d68-9fb2-76d2d03751b1)

### Please consider and list out some caveats or tradeoffs made in your design decision

The tradeoff is the fact that entering multiline items in your shopping list should be done with `shift + enter` instead of `enter`.
It is my opinion that that is acceptable, as in my case >99% of the items I add to a shopping list are simple entries of 1-3 words, without any need for newlines.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
Hey, thanks for the detailed feature request. 

This sounds pretty sensible to me, and seems like it would improve the ux of adding things to the shipping list manually quite a lot. 
As you stated that you are willing to work on this yourself i would say go for it. PRs are always welcome. 

If you have any questions or are stuck somewhere along the way, feel free to reach out. The discord would be propably be the fastest way to get a response from one of the developers around here.

**Comment by Fastjur** [comment]
Opened a PR: https://github.com/mealie-recipes/mealie/pull/3118
